packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -1261,33 +1369,22 @@ export class CrossChainMessenger {
       throw new Error(`state root for message not yet published`)
     }
 
-    const receipt = await this.l2Provider.getTransactionReceipt(
-      resolved.transactionHash
-    )
-
-    const [withdrawal, withdrawalHash] = this.getWithdrawalFromReceipt(
-      receipt,
-      resolved
-    )
-
-    // Sanity check
-    if (withdrawal.MessagePassed.withdrawalHash !== withdrawalHash) {
```

Comment:
Slight preference to use the hash from the event instead of computing the hash client side, so that we can "just work" with protocol upgrades